test/jdk/java/foreign/enablenativeaccess/TestEnableNativeAccessJarManifest.java

```diff
@@ -0,0 +1,97 @@
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @enablePreview
+ * @requires jdk.foreign.linker != "UNSUPPORTED"
+ * @requires !vm.musl
+ *
+ * @library /test/lib
+ * @build TestEnableNativeAccess
+ *        org.openjdk.foreigntest.PanamaMainUnnamedModule
+ * @run testng TestEnableNativeAccessJarManifest
+ * @summary Basic test for Enable-Native-Access attribute in the
+ *          manifest of a main application JAR
+ */
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.jar.Attributes;
+import java.util.jar.Manifest;
+
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
+import jdk.test.lib.util.JarUtils;
+
+import org.testng.annotations.Test;
+
+public class TestEnableNativeAccessJarManifest extends TestEnableNativeAccessBase {
+
+    /**
+     * Runs the test to execute the given test action. The VM is run with the
+     * given VM options and the output checked to see that it matches the
+     * expected result.
+     */
+    OutputAnalyzer run(String action, String cls, Result expectedResult, String... attributes) throws Exception {
+        Manifest man = new Manifest();
+        Attributes attrs = man.getMainAttributes();
+        attrs.put(Attributes.Name.MANIFEST_VERSION, "1.0");
+        attrs.put(Attributes.Name.MAIN_CLASS, cls);
+
+        for (String nameAndValue : attributes) {
+            String[] s = nameAndValue.split("=");
+            if (s.length != 2)
+                throw new RuntimeException("Malformed: " + nameAndValue);
+            String name = s[0];
+            String value = s[1];
+            attrs.put(new Attributes.Name(name), value);
+        }
+
+        // create the JAR file with Test1 and Test2
+        Path jarfile = Paths.get(action + ".jar");
+        Files.deleteIfExists(jarfile);
+
+        Path classes = Paths.get(System.getProperty("test.classes", ""));
+        JarUtils.createJarFile(jarfile, man, classes, Paths.get(cls.replace('.', '/') + ".class"));
+
+        // java -jar test.jar
+        OutputAnalyzer outputAnalyzer = ProcessTools.executeTestJava(
+                    "--enable-preview",
+                    "-Djava.library.path=" + System.getProperty("java.library.path"),
+                    "-jar", jarfile.toString())
+                .outputTo(System.out)
+                .errorTo(System.out);
+        checkResult(expectedResult, outputAnalyzer);
+        return outputAnalyzer;
+    }
+
+    @Test
+    public void testSucceed() throws Exception {
+        run("panama_no_unnamed_module_native_access", UNNAMED, successWithWarning("ALL-UNNAMED"), new String[]{});
```

Comment:
Maybe using a record to capture attribute name/value would avoid the need to split the string?
Also, other tests use a data provider, should we do that here too?